go/extractor/autobuilder/build-environment.go

```diff
@@ -273,14 +271,10 @@ func IdentifyEnvironment() {
 	defer project.RemoveTemporaryExtractorFiles()
 
 	// Find the greatest Go version required by any of the workspaces.
-	greatestGoVersion := project.RequiredGoVersion(&workspaces)
-	v.goModVersion, v.goModVersionFound = greatestGoVersion.Version, greatestGoVersion.Found
+	v.goModVersion = project.RequiredGoVersion(&workspaces)
 
 	// Find which, if any, version of Go is installed on the system already.
-	v.goEnvVersionFound = toolchain.IsInstalled()
-	if v.goEnvVersionFound {
-		v.goEnvVersion = toolchain.GetEnvGoVersion()[2:]
-	}
+	v.goEnvVersion = toolchain.GetEnvGoSemVer()
```

Comment:
We've lost a call to `toolchain.IsInstalled()`. Are you sure that that isn't needed? Do we have any tests that cover that case?

Reply:
Good point! Thanks for spotting this.

`toolchain.IsInstalled()` just looks up the path of the `go` executable with `exec.LookPath` and returns `true` or `false` depending on whether that succeeded. 

Meanwhile, `toolchain.GetEnvGoSemVer()` calls `toolchain.GetEnvGoVersion()` which invokes `go version`. Unlike `toolchain.IsInstalled()` this fails if something goes wrong.

So, if `go` is not installed, then previously we would have ended up with no `goEnvVersion`, but now that would actually terminate the process. So that's definitely a problem.

Reply:
With respect to tests, I have some work on testing `resolve build-environment` specifically. I will try to get that done ASAP as well.

Reply:
No need to expedite the tests - I just meant that you could add it to your list of things that need testing.

Reply:
The work on those tests is almost done (I did 99% of it a few weeks ago) and they would have caught this.

Reply:
Actually those tests wouldn't have caught it either, because there is no nice way to simulate `go` not being installed in an integration test. I played around with different approaches, but I am not too keen on any of them. Happy to discuss the options I tried with you.

I instead looked at migrating our old `resolve build-environment` tests from when we initially worked on the feature to the internal repo, and have a draft PR for that there now.

I am also wondering if `GetEnvGoSemVer` should really use `log.Fatalf` or if we could just get away with returning `nil` or throwing it at `util.NewSemVer` to see if it can make sense of it. We'd have to consider the downstream implications of doing that, though.

For now, I have addressed this specific issue by restoring the check for `toolchain.IsInstalled()`